src/metabase/embed/settings.clj

```diff
@@ -5,66 +5,105 @@
+
+(mu/defn- ignore-localhost :- :string
+  "Remove localhost:* or localhost:<port> from the list of origins."
+  [s :- [:maybe :string]]
+  (->> (str/split (or s "") #"\s+")
+       (remove #(re-matches #"localhost:(\*|\d+)" %))
+       distinct
+       (str/join " ")
+       str/trim))
+
+(mu/defn- add-localhost :- :string [s :- [:maybe :string]]
+  (->> s ignore-localhost (str "localhost:* ") str/trim))
+
+(defn embedding-app-origins-sdk-setter
+  "The setter for [[embedding-app-origins-sdk]].
+
+  Checks that we have SDK embedding feature and that it's enabled, then sets the value accordingly."
+  [new-value]
+  (add-localhost ;; return the same value that is returned from the getter
+   (when (and (premium-features/has-feature? :embedding-sdk)
+              ;; Cannot set the SDK origins if the SDK embedding is disabled. so it will remain localhost:*.
+              (setting/get-value-of-type :boolean :enable-embedding-sdk))
+     (->> new-value
+          ignore-localhost
+          (setting/set-value-of-type! :string :embedding-app-origins-sdk)))))
+
 (defsetting embedding-app-origins-sdk
   (deferred-tru "Allow this origin to embed Metabase SDK")
-  :feature    :embedding-sdk
+  :type       :string
   :export?    false
   :visibility :public
   :encryption :no
-  :audit      :getter)
-
-(defsetting enable-embedding
-  (deferred-tru "Allow admins to securely embed questions and dashboards within other applications?")
-  :type       :boolean
-  :default    false
-  :visibility :authenticated
-  :export?    true
   :audit      :getter
-  :setter     (fn [new-value]
-                (when (not= new-value (setting/get-value-of-type :boolean :enable-embedding))
-                  (setting/set-value-of-type! :boolean :enable-embedding new-value)
-                  (when (and new-value (str/blank? (embed/embedding-secret-key)))
-                    (embed/embedding-secret-key! (crypto-random/hex 32)))
-                  (snowplow/track-event! ::snowplow/embed_share
-                                         {:event                      (if new-value
-                                                                        :embedding-enabled
-                                                                        :embedding-disabled)
-                                          :embedding-app-origin-set   (boolean (embedding-app-origin))
-                                          :number-embedded-questions  (t2/count :model/Card :enable_embedding true)
-                                          :number-embedded-dashboards (t2/count :model/Dashboard :enable_embedding true)}))))
+  :getter    (fn embedding-app-origins-sdk-getter []
+               (add-localhost (setting/get-value-of-type :string :embedding-app-origins-sdk)))
+  :setter   embedding-app-origins-sdk-setter)
 
 (defsetting enable-embedding-static
   (deferred-tru "Allow admins to embed Metabase via static embedding?")
   :type       :boolean
+  :feature    :embedding
   :default    false
```

Comment:
I'm not sure we need this here anymore. Usually all of embedding was hidden behind the 'embedding' feature, but I'm fairly sure that static embedding isn't restricted to the premium feature

Reply:
Good call. Thank you!